/*
 * This file was automatically generated by EvoSuite
 * Fri Dec 30 09:19:40 GMT 2016
 */

package account;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import account.Account;
import account.BankAccount;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.System;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, useJEE = true) 
public class Account_ESTest extends Account_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[7];
      account0.checkResult(0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      account0.checkResult(0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.go(stringArray0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.go(stringArray0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.checkResult(0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.checkResult(0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      account0.checkResult(0);
      // Undeclared exception!
      try { 
        Account.Service(0, 0);
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("account.Account", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.go(stringArray0);
      Account.Service(0, 0);
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.checkResult(0);
      Account.Service(0, 0);
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
      Account.Service(0, 0);
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[2];
      account0.Bank_Total = (-26);
      // Undeclared exception!
      try { 
        account0.go(stringArray0);
      } catch(System.SystemExitException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.System", e);
      }
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      Account.Bank_random = null;
      Account account0 = new Account();
      String[] stringArray0 = new String[7];
      // Undeclared exception!
      try { 
        account0.go(stringArray0);
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      Account account0 = new Account();
      account0.NUM_ACCOUNTS = (-297);
      // Undeclared exception!
      try { 
        account0.go((String[]) null);
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("account.Account", e);
      }
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      BankAccount[] bankAccountArray0 = new BankAccount[3];
      Account.accounts = bankAccountArray0;
      // Undeclared exception!
      try { 
        Account.Service((-2), 2);
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // -2
         //
         verifyException("account.Account", e);
      }
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      Account account0 = new Account();
      // Undeclared exception!
      try { 
        account0.checkResult((-3114));
      } catch(System.SystemExitException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.System", e);
      }
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[0];
      account0.go(stringArray0);
      Account.Service(0, (-950));
  }
}
